### PR TITLE
More cleanup of the site dispatcher

### PR DESCRIPTION
**Pull request by live627**
This is my go at cleaning up one of the most important files in Elkarte.

A fool's work is never done, though.

### CODE REVIEW

**Comment by emanuele45** [comment]
Juts a technical remark: it would be better to avoid merges, but use rebase instead, if you have problems with it, I can do the rebase for you, I got quite used to them. lol

File: sources/SiteDispatcher.class.php [hunk 1]
```diff
@@ -88,6 +92,11 @@ public function __construct($action = null, $subaction = null, $area = null)
 		call_integration_hook('integrate_action_frontpage', array(&$this->_default_action));
 
 		$this->_noActionActions($action, !empty($modSettings['allow_guestAccess']));
```
Comment (Spuds) [inline sources/SiteDispatcher.class.php]: $action is not defined anylonger

File: sources/SiteDispatcher.class.php [hunk 2]
```diff
@@ -88,6 +92,11 @@ public function __construct($action = null, $subaction = null, $area = null)
...
+		// A safer way to work with our form globals
+		$_req = HttpReq::instance();
```
Comment (Spuds) [inline sources/SiteDispatcher.class.php]: With doing this, do we need to pass it ($-req) to the constructor any longer?

**Comment by emanuele45** [comment]
Apart from that I'd say... cool stuff! :dancer:

**Comment by Spuds** [comment]
LOL .. I made comments right while live627 was making the updates to the commit.

**Comment by live627** [comment]
> LOL .. I made comments right while live627 was making the updates to the commit.

Yeah... bad merge.

> it would be better to avoid merges, but use rebase instead

Why? To get rid of merge commits?

**Comment by emanuele45** [comment]
Sorry, I got the email notification only today... :-\ 

Yep, that and because the tree becomes cleaner.

**Comment by live627** [comment]
GH has the option to not make a merge commit in the repo options-I just now found this out accidentally.

**Comment by emanuele45** [comment]
GH has an option to squash all the commits in a PR into a single commit (squash and merge), what I was suggesting is a little different (rebase instead of the merge you did to keep up with the changes in development).
Anyway, this is already here, and we have done it in the past, so I guess I can avoid being too picky.

**Pull request merged**